conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -175,6 +180,54 @@ def requests_ca_bundle_check(prefix: str, verbose: bool) -> None:
             )
 
 
+def consistent_env_check(prefix: str, verbose: bool) -> None:
+    pm = context.plugin_manager  # get the plugin manager from context
+    SolverClass = (
+        pm.get_solver_backend()
+    )  # get the solver backend from the plugin manager
+    pd = PrefixData(prefix)  # get prefix data
+    repodatas = {}  # create a repodatas dict
+
+    # segregate the package records based on subdir/architecture and package type
+    for record in pd.iter_records():
+        record_data = dict(record.dump())
+        if record.fn.endswith(".conda"):
+            pkg_type = "packages.conda"
+        else:
+            pkg_type = "packages"
+
+        if record_data["subdir"] not in repodatas:
+            repodatas[record_data["subdir"]] = {}
+        if pkg_type not in repodatas[record_data["subdir"]]:
+            repodatas[record_data["subdir"]][pkg_type] = {}
+        repodatas[record_data["subdir"]][pkg_type][record.fn] = record_data
+
+    # create fake channel with package records from pd
+    with TemporaryDirectory() as tmp_dir:
+        for subdir in repodatas:
+            path = Path(tmp_dir) / subdir
+            path.mkdir(parents=True, exist_ok=True)
+            repodata = path / "repodata.json"
+            if repodatas[subdir]["packages.conda"]:
+                repodata.write_text(json.dumps(repodatas[subdir]["packages.conda"]))
+            if repodatas[subdir]["packages"]:
+                repodata.write_text(json.dumps(repodatas[subdir]["packages"]))
```

Comment:
Currently you are only writing `packages.conda` or `packages` to the `repodata.json`, we want both:

```suggestion
        for subdir, repodata in repodatas.items():
            path = Path(tmp_dir, subdir)
            path.mkdir(parents=True, exist_ok=True)
            (path / "repodata.json").write_text(json.dumps(repodata))
```

Reply:
@kenodegard, I am using two `if` statements and not `if else`.

Reply:
`write_text` clears the contents first, doesn't it?